modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/ConnectionProviderAwareMetadataContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.metadata;
+
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.metadata.MetadataContext;
+
+import java.util.Optional;
+
+/**
+ * A {@link MetadataContext} which is aware of the {@link ConnectionProvider} used to provide its connection.
+ *
+ * @since 4.4.0, 4.3.1
+ */
+public interface ConnectionProviderAwareMetadataContext extends MetadataContext {
```

Comment:
why the specialization intstead of adding this to the general metadataContext ?

Reply:
MetadataContext is an api interface that developers use when resolving metadata. We don't want them to access to the instance of the connection provider. This ConnectionProviderAwareMetadataContext is an internal interface.